### PR TITLE
Add apply filters btn

### PR DESCRIPTION
**Pull request by amiraabouhadid**
### Description

Remove watchers on filters, add an apply button instead

### Changes

- remove watchers for filter inputs in nodes, farms & dedicated nodes
- add applyFilters function for button

### Related Issues

[#1775](https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1775)

image
image
image

### CODE REVIEW

**Comment by ehab-hassan** [comment]
image
 - make 'clear' button at right and color="anchor"
 - apply color="secondary"
 - space between buttons maybe mr-2

**Comment by amiraabouhadid** [comment]
> image
> 
> * make 'clear' button at left and color="anchor"
> * Apply button color="secondary"
> * space between buttons maybe mr-2



> image
> 
> * make 'clear' button at left and color="anchor"
> * Apply button color="secondary"
> * space between buttons maybe mr-2
applied 
image

**Comment by ehab-hassan** [comment]
@amiraabouhadid make clear button at left and apply at right

**Comment by amiraabouhadid** [comment]
> @amiraabouhadid make clear button at left and apply at right

image

**Comment by ehab-hassan** [comment]
perfect thanks

**Comment by AlaaElattar** [comment]
- I think the buttons will look better at the left. What do u think ?

**Comment by ehab-hassan** [comment]
> > * I think the buttons will look better at the left. What do u think ?
> 
> image
> 

Agree100%

**Comment by AlaaElattar** [comment]
- Here i chose offline nodes, then i wanted to filter on them.
- but the filters applied on all nodes not offline .

[Screencast from 01-01-2024 03:04:36 PM.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/54944491/2c8af56d-7c83-4ae2-803f-b08e574435b6)

**Comment by AlaaElattar** [comment]
- The initial value is `up` in the Nodes page, it should contain all nodes (up, down, and standby) 

image

**Comment by maayarosama** [comment]
Initially the node status on nodes pages shouldn't be up, All nodes should be displayed

**Comment by AlaaElattar** [comment]
- When i reset filters, I expect the nodes status to be reset also.

[Screencast from 01-02-2024 11:30:00 AM.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/54944491/76034259-dcd3-4734-893d-83b1dbae5f05)

**Review by AlaaElattar — APPROVED** [review]
works fine

File: packages/playground/src/components/filter.vue [hunk 1]
```diff
@@ -1,69 +1,82 @@
...
+            </form-validator>
+          </v-row>
+
+          <v-row>
+            <v-col cols="12" sm="4" md="2" class="d-flex justify-start align-center mb-6 ml-auto mr-4">
               <v-btn
-                :disabled="!isFiltersTouched || formDisabled"
+                class="mr-4"
+                :disabled="!isValidForm"
+                :loading="loading"
```
Comment (0oM4R) [inline packages/playground/src/components/filter.vue:54]: i think we need to set it to disabled not loading
image

File: packages/playground/src/views/nodes.vue [hunk 2]
```diff
@@ -158,20 +163,25 @@ export default {
 
     const request = debounce(_requestNodes, 1000);
```
Comment (0oM4R) [inline packages/playground/src/views/nodes.vue:164]: i think there is no need for the denounce now

**Review by 0oM4R — CHANGES_REQUESTED** [review]

File: packages/playground/src/views/nodes.vue [hunk 3]
```diff
@@ -194,6 +204,7 @@ export default {
 
     onMounted(async () => {
       await checkSelectedNode();
+      // filterOptions.value = optionsInitializer();
```
Comment (0oM4R) [inline packages/playground/src/views/nodes.vue:207]: please remove the comment if it is not needed

**Comment by 0oM4R** [comment]
clear btn is still loading please check it out

**Comment by 0oM4R** [comment]
please check network on each click will send 2 requests
image

**Comment by AlaaElattar** [comment]
- clear button shouldn't be enabled until user starts to filter 

image

**Review by mohamedamer453 — COMMENTED** [review]
- At small window sizes the Apply button exceeds the bounds of the filters box.

   image

**Comment by AlaaElattar** [comment]
- when i filter after choosing node status, the node status is cleared. 

[Screencast from 01-02-2024 03:16:18 PM.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/54944491/55556d99-96f5-4dd6-82cb-b384b5ac3382)

**Review by AlaaElattar — APPROVED** [review]
Great Job

**Comment by Mahmoud-Emad** [comment]
When enabling the GPU or Gateway options and then pressing on apply button, the switches go back to being false

[Screencast from 01-03-2024 09:45:50 AM.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/57001890/e1713af4-efa6-4148-8165-748bed06f415)

**Comment by Mahmoud-Emad** [comment]
The GPU option in the Dedicated Nodes screen doesn't work as expected.

[Screencast from 01-03-2024 09:50:24 AM.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/57001890/ccd18b0c-39a3-4995-a438-d725c17246fc)

**Comment by Mahmoud-Emad** [comment]
When filtering with values e.g. Node ID, when I want to display all of the nodes the button of `Apply` gets disabled and I can't go back.

[Screencast from 01-03-2024 09:54:53 AM.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/57001890/b586e95e-2c1d-453f-aedb-3cb2b50e92fa)

**Comment by Mahmoud-Emad** [comment]
> The GPU option in the Dedicated Nodes screen doesn't work as expected.
> 
>  Screencast.from.01-03-2024.09.50.24.AM.webm 
> No GPU card on the node 148 as seen in the screen
> 
> image

The resolution of this comment introduces a new concern. The existing functionality for the grid proxy's GPU filter is set to "True," indicating a requirement for GPU, while "False" signifies the absence of this requirement. Consequently, when the switch is set to "False," it implies the necessity to filter all nodes with GPU filter set to "False." Consequently, nodes without GPUs will not be returned. 

Refer to the [Screencast from 01-03-2024 12:51:37 PM.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/57001890/993e36dd-933b-4202-9466-e8cc4d7258b6) for more clarity.

To address this issue, you have two potential solutions:
1. Rectify the situation in the grid proxy.
2. Resolve it by substituting the "False" keyword with "undefined" :D

**Review by MohamedElmdary — APPROVED** [review]

**Pull request merged**